NullPointerException with HibernateValidator @Valid

**Describe the bug**

Validator annotations can be placed to various places. Example this is probably not effective:

public class StringBasedIdentity implements Serializable, Comparable<StringBasedIdentity> {

    private static final long serialVersionUID = 1L;

    @NotNull
    @NotEmpty
    private final String name;

...

}


**Expected behavior**
 Quarkus processor should be able to pinpoint the error with at least the location of the annotation should be provided as a hint.

**Actual behavior**
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.2.1.Final:build (default) on project sched-schedule-app: Failed to build a runnable JAR: Failed to augment application classes: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.hibernate.validator.deployment.HibernateValidatorProcessor#build threw an exception: java.lang.NullPointerException
[ERROR]         at io.quarkus.hibernate.validator.deployment.HibernateValidatorProcessor.build(HibernateValidatorProcessor.java:191)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:915)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:279)
[ERROR]         at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
[ERROR]         at java.lang.Thread.run(Thread.java:748)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:479)


**To Reproduce**
Steps to reproduce the behavior:
1. Create file
2. Run build

**Environment (please complete the following information):**
 - Quarkus version 1.2.1

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -15,7 +15,7 @@
 
     <properties>
         <aesh.version>1.11</aesh.version>
-        <jandex.version>2.1.2.Final</jandex.version>
+        <jandex.version>2.1.3.Final</jandex.version>
         <resteasy.version>4.5.2.Final</resteasy.version>
         <opentracing.version>0.31.0</opentracing.version>
         <opentracing-jaxrs.version>0.4.1</opentracing-jaxrs.version>

--- a/independent-projects/arc/pom.xml
+++ b/independent-projects/arc/pom.xml
@@ -37,7 +37,7 @@
         <!-- Versions -->
         <version.cdi>2.0.2</version.cdi>
         <version.jta>1.3.3</version.jta>
-        <version.jandex>2.1.2.Final</version.jandex>
+        <version.jandex>2.1.3.Final</version.jandex>
         <version.junit5>5.5.2</version.junit5>
         <version.maven>3.5.4</version.maven>
         <version.assertj>3.15.0</version.assertj>

